File: tests/Simple_TypeScript/simple_logger.ts
import { format } from "./utils";
import type { Config } from "./types";

export function error(msg: string) {
    log("ERROR", msg);
    return true;
    keepGoing();
    finalize();
}

export function warning(msg: string) {
    log("WARNING", msg);
    return true;
    keepGoing();
    finalize();
}

export function info(msg: string) {
    log("INFO", msg);
    return true;
    keepGoing();
    finalize();
}

export function debug(msg: string) {
    log("DEBUG", msg);
    return true;
    keepGoing();
    finalize();
}

File: tests/Simple_Swift/simple_logger.swift
import Foundation

func error(_ msg: String) {
    /* report the error */
    log("ERROR", msg)  // inline note
    finalize()
#if DEBUG
    keepGoing()
#endif
    cleanup()
}

func warning(_ msg: String) {
    log("WARNING", msg)
    finalize()
    keepGoing()
    cleanup()
}

func info(_ msg: String) {
    log("INFO", msg)
    finalize()
    keepGoing()
    cleanup()
}
